src/iff.imageio/iffinput.cpp

```diff
@@ -526,210 +593,391 @@ IffInput::readimg()
+                if (!ioread(scratch.data(), 1, scratch.size())) {
                     return false;
+                }
 
-                // set tile data
-                uint8_t* p = static_cast<uint8_t*>(&scratch[0]);
+                span<uint8_t> scratch_span(scratch);
 
-                if (tile_compress) {
-                    // set map
+                if (tile_compressed) {
                     std::vector<uint8_t> map;
                     if (littleendian()) {
-                        int rgb16[]  = { 0, 2, 4, 1, 3, 5 };
-                        int rgba16[] = { 0, 2, 4, 6, 1, 3, 5, 7 };
-                        if (m_iff_header.pixel_channels == 3) {
-                            map = std::vector<uint8_t>(rgb16, &rgb16[6]);
-                        } else {
-                            map = std::vector<uint8_t>(rgba16, &rgba16[8]);
-                        }
-
+                        uint8_t rgb16[]  = { 0, 2, 4, 1, 3, 5 };
+                        uint8_t rgba16[] = { 0, 2, 4, 6, 1, 3, 5, 7 };
+                        map              = (m_header.rgba_count == 3)
+                                               ? std::vector<uint8_t>(rgb16, rgb16 + 6)
+                                               : std::vector<uint8_t>(rgba16, rgba16 + 8);
                     } else {
-                        int rgb16[]  = { 1, 3, 5, 0, 2, 4 };
-                        int rgba16[] = { 1, 3, 5, 7, 0, 2, 4, 6 };
-                        if (m_iff_header.pixel_channels == 3) {
-                            map = std::vector<uint8_t>(rgb16, &rgb16[6]);
-                        } else {
-                            map = std::vector<uint8_t>(rgba16, &rgba16[8]);
-                        }
+                        uint8_t rgb16[]  = { 1, 3, 5, 0, 2, 4 };
+                        uint8_t rgba16[] = { 1, 3, 5, 7, 0, 2, 4, 6 };
+                        map              = (m_header.rgba_count == 3)
+                                               ? std::vector<uint8_t>(rgb16, rgb16 + 6)
+                                               : std::vector<uint8_t>(rgba16, rgba16 + 8);
                     }
 
-                    // map BGR(A)BGR(A) to RRGGBB(AA)
-                    for (int c = (channels * m_spec.channel_bytes()) - 1;
+                    for (int c = m_header.rgba_count * m_header.channel_bytes()
+                                 - 1;
                          c >= 0; --c) {
                         int mc = map[c];
 
                         std::vector<uint8_t> in(tw * th);
-                        uint8_t* in_p = &in[0];
+                        span<uint8_t> in_span(in);
+
+                        size_t used = uncompress_rle_channel(scratch_span,
+                                                             in_span, tw * th);
+                        if (used > scratch_span.size()) {
+                            errorfmt(
+                                "RLE uncompress exceeds span size (channel byte %d)",
```

Comment:
All the functions ending in `fmt` take the new std::format idioms, so this should be:

```suggestion
                                "RLE uncompress exceeds span size (channel byte {})",
```